src/mcmc/Inference.jl

```diff
@@ -98,19 +98,15 @@ Wrap a sampler so it can be used as an inference algorithm.
 """
 externalsampler(sampler::AbstractSampler) = ExternalSampler(sampler)
 
-"""
-    ESLogDensityFunction
-
-A log density function for the External sampler.
-
-"""
-const ESLogDensityFunction{M<:Model,S<:Sampler{<:ExternalSampler},V<:AbstractVarInfo} = Turing.LogDensityFunction{V,M,<:DynamicPPL.DefaultContext}
-function LogDensityProblems.logdensity(f::ESLogDensityFunction, x::NamedTuple)
+function LogDensityProblems.logdensity(
+    f::Turing.LogDensityFunction{<:AbstractVarInfo,<:Model,<:DynamicPPL.DefaultContext},
+    x::NamedTuple
+)
```

Comment:
This is type piracy (I know it was already there 😉 - and BTW `DynamicPPL.unflatten` below is another instance of type piracy). In DynamicPPL, we explicitly decided to not add this overload: https://github.com/TuringLang/DynamicPPL.jl/pull/501 Can we just remove it completely?

Reply:
Yeaaah I'm aware of both and I don't like them :confused: 

It's done to be compatible with AdvancedMH.jl which allows you to specify the proposal distribution using `NamedTuple` :confused: So we can't just remove it without messing with this; we'd have to do some more work for that, unfortunately.

Reply:
Ohh haha I missed that, I just remembered that the issue had been closed 😅

Reply:
Yes, let's address this separately.